vault/src/metrics.rs

```diff
@@ -110,6 +112,18 @@ pub struct PerCurrencyMetrics {
     utxo_count: IntGauge,
 }
 
+#[async_trait]
+pub trait VaultDataReader<BCA: BitcoinCoreApi + Clone + Send + Sync + 'static> {
+    async fn get_entries(&self) -> Vec<VaultData<BCA>>;
+}
```

Comment:
I defined this trait so I could mock the `VaultIdManager` in the tests

Reply:
Not a huge fan of that tbh. Afaik VaultIdManager should already be mockable by using `VaultIdManager::from_map` and passing in a mock `parachain_rpc`

Reply:
Yeah agree this also doesn't sit quite right with me, @savudani8 can you check @sander2's suggestion?

Reply:
I think this is still the cleanest approach. I tried the suggestion, but mocking `parachain_rpc` requires mocking `on_event`, which is a generic function. Moving `on_event` to the `UtilFuncs` trait to mock it should've worked, but there is a [bug](https://github.com/asomers/mockall/issues/299) in `mockall`, which requires adding the `'static` bound. This would add a lot of changes itself so isn't really an option. As a workaround to adding `'static` bounds everywhere, I tried keeping the current `UtilFuncs` as `#[cfg(not(test))]` while adding a test version with `'static` bounds, which also didn't work.

Reply:
Hmm yea that's unfortunate.. I guess we go with the original approach then